lib/src/screens/buy/onramper_page.dart

```diff
@@ -70,17 +63,30 @@ class OnRamperPageBody extends StatefulWidget {
 class OnRamperPageBodyState extends State<OnRamperPageBody> {
   OnRamperPageBodyState();
 
-  @override
-  void initState() {
-    super.initState();
-    if (Platform.isAndroid) WebView.platform = SurfaceAndroidWebView();
-  }
-
   @override
   Widget build(BuildContext context) {
-    return WebView(
-        initialUrl: widget.uri.toString(),
-        backgroundColor: widget.backgroundColor,
-        javascriptMode: JavascriptMode.unrestricted);
+    return InAppWebView(
+      initialOptions: InAppWebViewGroupOptions(
+        crossPlatform: InAppWebViewOptions(transparentBackground: true),
+      ),
+      initialUrlRequest: URLRequest(
+        url: Uri.tryParse(
+          widget.uri.toString(),
+        ),
+      ),
+      androidOnPermissionRequest: (_, __, resources) async {
+        bool permissionNotGranted = await Permission.camera.status != PermissionStatus.granted;
+        if (permissionNotGranted) {
+          permissionNotGranted = await Permission.camera.request().isGranted;
```

Comment:
if the user granted the permission so this will return `true` to the `permissionNOTgranted` variable, and then you will check on it and if it's true you will set the response action to `DENY` which should be `GRANT`